Client/Frontend/Browser/BrowserToolbar.swift

```diff
@@ -36,102 +29,72 @@ class BrowserToolbar: UIView, UITextFieldDelegate, BrowserLocationViewDelegate {
 
     override init(frame: CGRect) {
         super.init(frame: frame)
-        viewDidInit()
+        initViews()
     }
 
     required init(coder aDecoder: NSCoder) {
         super.init(coder: aDecoder)
-        viewDidInit()
     }
 
-    private func viewDidInit() {
-        self.backgroundColor = UIColor(white: 0.80, alpha: 1.0)
+    func addButtons(buttons: UIButton...) {
+        for button in buttons {
+            button.setTitleColor(UIColor.blackColor(), forState: UIControlState.Normal)
+            button.setTitleColor(UIColor.grayColor(), forState: UIControlState.Disabled)
+            button.imageView?.contentMode = UIViewContentMode.ScaleAspectFit
+            addSubview(button)
+        }
+
+        var prev: UIView? = nil
+        for view in self.subviews {
+            if let view = view as? UIView {
+                view.snp_remakeConstraints { make in
+                    if let prev = prev {
+                        make.left.equalTo(prev.snp_right)
+                    } else {
+                        make.left.equalTo(self)
+                    }
+                    prev = view
+
+                    make.centerY.equalTo(self)
+                    make.height.equalTo(ToolbarHeight - DefaultPadding * 2)
+                    make.width.equalTo(self).dividedBy(self.subviews.count)
+                }
+            }
+        }
+    }
+
+    private var inited = false
+    func initViews() {
+        if inited {
```

Comment:
Er, what's this for? If we have a problem with <code>initViews</code> being called multiple times, we should fix that problem instead of papering around it with a boolean.